### PR TITLE
[LO-3696] Filter sensitive data from query params

### PR DESCRIPTION
**Pull request by biggless**
Filter sensitive data from query params & url.

### CODE REVIEW

File: lib/plugsnag/basic_error_report_builder.ex [hunk 1]
```diff
@@ -55,6 +55,25 @@ defmodule Plugsnag.BasicErrorReportBuilder do
     |> Keyword.get(field, [])
   end
 
+  defp filter(:query_string, data) when is_binary(data) do
+    {_, data} = data
+    |> String.split("&")
```
Comment (stephanos) [inline lib/plugsnag/basic_error_report_builder.ex]: have you considered using something like https://hexdocs.pm/elixir/URI.html#decode_query/2 to do the decoding for you?

**Comment by stephanos** [comment]
do you think it'd be possible to add a test for this? if we do that, it might increase the chances of having this merged back into the original project.

File: lib/plugsnag/basic_error_report_builder.ex (hunk 1 above)
Comment (biggless) [reply lib/plugsnag/basic_error_report_builder.ex]: Didn't know about this func, it simplifies a lot! Thanks!

**Comment by biggless** [comment]
> do you think it'd be possible to add a test for this? if we do that, it might increase the chances of having this merged back into the original project.

I'll add testing.

File: lib/plugsnag/basic_error_report_builder.ex [hunk 2]
```diff
@@ -55,6 +55,12 @@ defmodule Plugsnag.BasicErrorReportBuilder do
     |> Keyword.get(field, [])
   end
 
+  defp filter(:query_string, data) when is_binary(data) do
+    data
+    |> URI.decode_query
+    |> do_filter(filters_for(:params))
+    |> URI.encode_query
```
Comment (joelcrocker) [inline lib/plugsnag/basic_error_report_builder.ex:62]: Style: [use parentheses for one-arity functions when using the pipe operator](https://github.com/christopheradams/elixir_style_guide#parentheses-pipe-operator).

Comment (biggless) [reply lib/plugsnag/basic_error_report_builder.ex]: Done.

**Pull request merged**